### PR TITLE
fix: :lady_beetle: KFI data are not showing on the KleverScan

### PR DESCRIPTION
**Pull request by saviojks**

### CODE REVIEW

File: src/utils/hooks/index.tsx [hunk 1]
```diff
@@ -117,6 +117,8 @@ export const useFetchPartial = <T,>(
           ) {
             setLoading(true);
             if (type !== 'assets') {
+              query = { ...query };
```
Comment (samufacanha2) [inline src/utils/hooks/index.tsx:120]: unnecessary spread

**Review by samufacanha2 — DISMISSED** [review]

File: src/utils/hooks/index.tsx (hunk 1 above)
Comment (saviojks) [reply src/utils/hooks/index.tsx:120]: image

**Review by samufacanha2 — APPROVED** [review]

File: src/pages/validators/index.tsx [hunk 2]
```diff
@@ -139,26 +138,29 @@ const Validators: React.FC<PropsWithChildren> = () => {
...
-      onChange: async value => {
-        setLoading(true);
-        await fetchPartialValidator(value);
+  const filters: IFilter[] = useMemo(() => {
+    return [
+      {
+        title: 'Name',
+        data: filterValidators
+          .map(validator => validator.name)
+          .filter(validator => !!validator) as string[],
```
Comment (lucasportella) [inline src/pages/validators/index.tsx:147]: why not just .filter(validator => validator.name)

File: src/services/requests/home/index.ts [hunk 3]
```diff
@@ -477,9 +477,13 @@ const homeKfiPriceCall = async (): Promise<
     });
 
     if (!res.error || res.error === '') {
-      const data = res?.data?.prices?.Exchanges.find(
-        (exchange: any) => exchange.ExchangeName === 'Klever',
-      );
+      const data =
+        res?.data?.prices?.Exchanges.find(
+          (exchange: any) => exchange.ExchangeName === 'Klever',
```
Comment (lucasportella) [inline src/services/requests/home/index.ts:482]: dont need 2 loops here just use the checks together exchange.ExchangeName === 'Klever' || !!exchange.ExchangeName

File: src/pages/validators/index.tsx [hunk 4]
```diff
@@ -139,26 +138,29 @@ const Validators: React.FC<PropsWithChildren> = () => {
...
-        if (value === 'All') {
-          setQueryAndRouter({}, router);
-        } else {
-          setQueryAndRouter({ name: value }, router);
-        }
-      },
-      onChange: async value => {
-        setLoading(true);
-        await fetchPartialValidator(value);
+  const filters: IFilter[] = useMemo(() => {
```
Comment (lucasportella) [inline src/pages/validators/index.tsx:141]: use useMemo only when calculation is expensive
https://react.dev/reference/react/useMemo#skipping-expensive-recalculations

File: src/pages/validators/index.tsx (hunk 2 above)
Comment (saviojks) [reply src/pages/validators/index.tsx:147]: image
image

**Pull request merged**